test/tests/LayoutGrids.js

```diff
@@ -0,0 +1,755 @@
+'use strict';
+
+const { ariaTest } = require('..');
+const { By, Key } = require('selenium-webdriver');
+
+const confirmAriaLabelledby = require('../util/confirmAriaLabelledby');
+const confirmAttributeValue = require('../util/confirmAttributeValue');
+
+const reload = async (session) => {
+  return session.get(await session.getCurrentUrl());
+};
+
+const clickUntilDisabled = async (session, selector) => {
+  const el = await session.findElement(By.css(selector));
+  await el.click();
+
+  if (await el.isEnabled()) {
+    return clickUntilDisabled(session, selector);
+  }
+};
+
+const checkActiveElement = function (/* gridcellsSelector, index */) {
+  let gridcellsSelector = arguments[0];
+  let index = arguments[1];
+  let gridcells = document.querySelectorAll(gridcellsSelector);
+  let gridcell = gridcells[index];
+  return (document.activeElement === gridcell) || gridcell.contains(document.activeElement);
+};
+
+const findFocusable = function (/* selector */) {
+  const selector = arguments[0];
+  const original = document.activeElement;
+  const focusable = Array.from(document.querySelectorAll(selector))
+    .map((parent) => {
+      const candidates = [parent, ...parent.querySelectorAll('*')];
+      for (let candidate of candidates) {
+
+        candidate.focus();
```

Comment:
I generally think this is okay, but we will at least want to validate it in the testing environment (see my comment on 56). The alternative might be to keep a list of focusable selectors, but that quickly gets out of hand when you consider things like `button:not(disabled):not([tabindex="-1"])`.

Reply:
I think @sh0ji is right here. I think the direction we want to take is to encode a list of focusable selectors. This matches the direction we got from @mcking65 last week as well.

The purpose here was to mutate the doc, actually, @jessebeach -- layout grid has some complicate logic for which element in a gridcell will take focus and, therefore, except keyboard interactions. The point of this function is to find AND focus on the focusable element in the gridcell. Also re: comment on 45, yup -- focus will not move to elements that are not focusable in these cases.

I'll switch this complicated logic over to a list of focusable elements based on the specific examples. I think it will be much clearer to read and understand and matches the direction we are going with other tests. Can I make a subtask for this and get to it this week -- and merge this PR for now, to unblock the landing of the util function refactor #787 which changes some code in this PR?

Reply:
Sure, I'll merge this now and look for the update later this week.

Reply:
> The purpose here was to mutate the doc, actually, @jessebeach

Ok, good to know. Thank you for the explanation.